### ISSUE TITLE
CPython support

### ISSUE THREAD

**Issue by penberg** [tursodatabase/limbo#128]
This is going to be a huge task, but the goal is to be able to build CPython using Limbo's SQLite compatibility library.

You can build Python with Limbo as follows:

```
git clone --depth 1 https://github.com/python/cpython.git
cd python
export LIBSQLITE3_LIBS=<LIMBO SOURCES>/target/debug/liblimbo_sqlite3.a
./configure
make -j8
```

I already added bunch of stubs to make it link, but there's tons of stuff to implement:

https://github.com/penberg/limbo/commit/17137ec458a9eed0eddc16d21e7ea457b057be36

**Comment by penberg** [comment]
After pull request #176, CPython no builds. The next step is to make it actually work:

```
penberg@turing:~/src/penberg/cpython$ ./python 
Python 3.14.0a0 (heads/main:e8c91d9, Jul 17 2024, 13:04:45) [GCC 13.2.1 20231205 (Red Hat 13.2.1-6)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sqlite3
>>> con = sqlite3.connect("tutorial.db")
thread '<unnamed>' panicked at sqlite3/src/lib.rs:339:5:
not yet implemented
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 5
Aborted (core dumped)
```

**Comment by lietu** [comment]
It seems unlikely people would want a custom build of Python to use a library, is there work in progress on a `pylimbo` -library that would specifically wrap this library and get published on PyPI?

**Comment by penberg** [comment]
@lietu Right, the main reason to do CPython support like this is more about ensuring SQLite compatibility. A `pylimbo` package makes sense when we have some more of the functionality covered. It should be pretty easy to build with something like https://github.com/PyO3/maturin

**Comment by penberg** [comment]
Not critical for 1.0 so moving out of milestone.

**Comment by roboturso** [comment]
This issue was triaged by @roboturso and determined to be out of scope: This is a feature/porting effort to add CPython support, not a report of a runtime failure to reproduce.
